### PR TITLE
HPCC-27386 Clearup Thor temp file usage

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by richardkchapman — CHANGES_REQUESTED** [review]
Appears to have revealed an issue with temp file names clashing when multiple jobs run in parallel

**Review by richardkchapman — APPROVED** [review]

File: thorlcr/slave/thslavemain.cpp [hunk 1]
```diff
@@ -116,11 +116,13 @@ static bool RegisterSelf(SocketEndpoint &masterEp)
         mySlaveNum = (unsigned)processGroup->rank(queryMyNode());
         assertex(NotFound != mySlaveNum);
         mySlaveNum++; // 1 based;
-        unsigned configSlaveNum = globals->getPropInt("@slavenum", NotFound);
-        if (NotFound != configSlaveNum)
-            assertex(mySlaveNum == configSlaveNum);
 
+        unsigned configSlaveNum = globals->getPropInt("@slavenum", NotFound);
```
Comment (mckellyln) [inline thorlcr/slave/thslavemain.cpp:120]: If we use channelsPerSlave, is the slavenum still always unique ?  I think so but wanted to check.

Comment (mckellyln) [inline thorlcr/slave/thslavemain.cpp:120]: If we use channelsPerSlave, is the slavenum still always unique ?  I think so but wanted to check.

Comment (jakesmith) [reply thorlcr/slave/thslavemain.cpp:120]: yes

File: thorlcr/slave/thslavemain.cpp [hunk 2]
```diff
@@ -583,7 +576,6 @@ int main( int argc, const char *argv[]  )
 #ifndef _CONTAINERIZED
     stopPerformanceMonitor();
 #endif
-    ClearTempDirs();
```
Comment (mckellyln) [inline thorlcr/slave/thslavemain.cpp:586]: So we are doing this in endJob now ?
What if anything in that try throws ?  Should we ClearTempDirs() outside the catch ?

**Review by mckellyln — COMMENTED** [review]
Looks good, just one more minor question about where ClearTempDirs() happens.

File: thorlcr/slave/thslavemain.cpp (hunk 2 above)
Comment (jakesmith) [reply thorlcr/slave/thslavemain.cpp:586]: @mckellyln - ClearTempDir does handle exceptions (and logs only) won't throw, so it's fine where it is.

**Review by mckellyln — APPROVED** [review]
Looks good.

**Comment by mckellyln** [comment]
Good to merge now

**Pull request merged**